core/src/main/java/org/ehcache/core/EhcacheBase.java

```diff
@@ -0,0 +1,504 @@
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void remove(K key) throws CacheWritingException {
+    removeInternal(key); // ignore return value;
+  }
+
+  protected abstract boolean removeInternal(final K key);
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void clear() {
+    clearObserver.begin();
+    statusTransitioner.checkAvailable();
+    try {
+      store.clear();
+      clearObserver.end(ClearOutcome.SUCCESS);
+    } catch (StoreAccessException e) {
+      clearObserver.end(ClearOutcome.FAILURE);
+      resilienceStrategy.clearFailure(e);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Iterator<Entry<K, V>> iterator() {
+    statusTransitioner.checkAvailable();
+    return new CacheEntryIterator(false);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<K, V> getAll(Set<? extends K> keys) throws BulkCacheLoadingException {
+    return getAllInternal(keys, true);
+  }
+
+  protected abstract Map<K,V> getAllInternal(Set<? extends K> keys, boolean b);
+
+
+  protected boolean newValueAlreadyExpired(K key, V oldValue, V newValue) {
+    return newValueAlreadyExpired(logger, runtimeConfiguration.getExpiryPolicy(), key, oldValue, newValue);
+  }
+
+  protected static <K, V> boolean newValueAlreadyExpired(Logger logger, ExpiryPolicy<? super K, ? super V> expiry, K key, V oldValue, V newValue) {
+    if (newValue == null) {
+      return false;
+    }
+
+    Duration duration;
+    try {
+      if (oldValue == null) {
+        duration = expiry.getExpiryForCreation(key, newValue);
+      } else {
+        duration = expiry.getExpiryForUpdate(key, supplierOf(oldValue), newValue);
+      }
+    } catch (RuntimeException re) {
+      logger.error("Expiry computation caused an exception - Expiry duration will be 0 ", re);
+      return true;
+    }
+
+    return Duration.ZERO.equals(duration);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public CacheRuntimeConfiguration<K, V> getRuntimeConfiguration() {
+    return runtimeConfiguration;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void init() {
+    statusTransitioner.init().succeeded();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void close() {
+    statusTransitioner.close().succeeded();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Status getStatus() {
+    return statusTransitioner.currentStatus();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void addHook(LifeCycled hook) {
+    statusTransitioner.addHook(hook);
+  }
+
+  void removeHook(LifeCycled hook) {
+    statusTransitioner.removeHook(hook);
+  }
+
+  protected void addBulkMethodEntriesCount(BulkOps op, long count) {
+    bulkMethodEntries.get(op).add(count);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<BulkOps, LongAdder> getBulkMethodEntries() {
+    return bulkMethodEntries;
+  }
+
+  protected static void checkNonNull(Object thing) {
```

Comment:
Can't we leverage Java 8 constructs for these?

Reply:
Yes